src/cryptonote_basic/cryptonote_basic.h

```diff
@@ -203,14 +203,39 @@ namespace cryptonote
     END_KV_SERIALIZE_MAP()
   };
 
+  // container for RTA identities (public keys)
+  // stores RTA payment ID, PoS public one-time identification key (used to identify PoS in the network and protect data for it),
+  // auth sample supernode public identification keys (graftnode will need it to validate auth sample signatures),
+  // PoS and Wallet Proxy Supernode identification keys to transaction_header.extra.
+  // TODO: better name?
+  struct rta_header
+  {
+    std::string payment_id;
+    std::vector<crypto::public_key> keys;
```

Comment:
I think we need to have the possibility to determine PoS public one-time identification key, PoS and Wallet Proxy Supernode public identification keys, since they are required for the transaction validation and we need to be sure that they are present.
We can store them in separate fields or use fixed indexes for adding them into the list.